src/global.cpp

```diff
@@ -38,38 +38,40 @@ void AssertWithMessage(bool condition, const char* message)
     }
 }
 
-void sm_crash( const char *reason )
+// sm_crash helper methods
+void HandleDebugBreak()
 {
-#if ( defined(_WIN32) && defined(CRASH_HANDLER) ) || defined(MACOSX) || defined(_XDBG)
-	/* If we're being debugged, throw a debug break so it'll suspend the process. */
-	if( IsDebuggerPresent() )
-	{
-		DebugBreak();
-		for(;;); /* don't return */
-	}
+    // If this is a debug build, use standard library assert. Otherwise, do nothing.
+#ifdef DEBUG
+    assert(false && "Debug break triggered by sm_crash");
 #endif
+}
 
+void ForceCrashHandler(const char* reason)
+{
 #if defined(CRASH_HANDLER)
 	CrashHandler::ForceCrash( reason );
 #else
-	std::abort();
-
-	/* This isn't actually reached.  We just do this to convince the compiler that the
-	 * function really doesn't return. */
-	for(;;);
+    std::terminate();
 #endif
+}
 
-#if defined(_WIN32)
-	/* Do something after the above, so the call/return isn't optimized to a jmp; that
-	 * way, this function will appear in backtrace stack traces. */
-#if defined(_MSC_VER)
-	__nop();
-#endif
+void AbortProcess()
+{
+#if defined(_WIN32) && defined(_MSC_VER)
+    __nop(); // ensure the function appears in backtrace stack traces
 #else
 	_exit( 1 );
 #endif
 }
 
+void sm_crash(const char* reason)
+{
+    HandleDebugBreak();
+    ForceCrashHandler(reason);
+    AbortProcess();
+}
```

Comment:
should we refactor sm_crash to use `try`/`catch` methodology in order to get more descriptive `crashinfo.txt`?